src/Camille.RiotGames/src/Util/RegionalRequester.cs

```diff
@@ -67,6 +97,18 @@ public RegionalRequester(IRiotGamesApiConfig config, string route)
             var retries = 0;
             var num429s = 0;
             var num5xxs = 0;
+
+            // If the region is not being used as a subdomain
+            if (_config.ApiCallRegionConfig == RegionConfig.InUrlAsRegionQueryParameter)
+            {
+                // Append the region as a query parameter, with consideration for other parameters
+                var uri = request.RequestUri
+                          + (!request.RequestUri.ToString().Contains("?") ? "?" : "&")
+                          + $"region={_region}";
```

Comment:
Would be better to do something like https://stackoverflow.com/a/14517976/2398020 to ensure everything is properly escaped (wow microsoft, this is really clunky)

Reply:
Yes, it is clunky ha - made only clunkier by System.Web's "absence" in a few of the target frameworks.
I explain further in a comment below.